src/Plot.elm

```diff
@@ -0,0 +1,107 @@
+module Plot where
+
+import Svg exposing (svg, Svg)
+import Private.Extras.SvgAttributes exposing (width, height)
+import Private.Dimensions exposing (Dimensions)
+import Private.Margins as Margins exposing (Margins)
+import Private.BoundingBox as BoundingBox exposing (BoundingBox)
+import Private.Scale.Utils as Scale
+import Private.Scale exposing (Scale)
+import Plot.Axis as Axis exposing (Axis)
+import Private.Bars as Bars exposing (Orient)
+import Private.Axis.View as AxisView
+import Private.Title as Title
+
+type alias Plot =
+  { dimensions: Dimensions
+  , margins: Margins
+  , svgs: List (BoundingBox -> List Svg)
+  , eventHandlers: List (BoundingBox -> Svg.Attribute)
+  , attrs: List Svg.Attribute
+  , title : Title.Model
+  }
+
+createPlot : Float -> Float -> Plot
+createPlot w h =
+  { dimensions = { width = w, height = h }
+  , margins = Margins.init
+  , svgs = []
+  , eventHandlers = []
+  , attrs = [width w, height h]
+  , title = Title.init
+  }
```

Comment:
It may be useful to add a field for scales, a method for `addScales` and then reuse the scales from the record throughout the functions. The upside is less arguments and a more consistent usage. I don't think you'd want a user to inadvertently pass in the wrong scales to a plot function.